llvm/lib/Transforms/Vectorize/LoopVectorizationLegality.cpp

```diff
@@ -943,11 +955,24 @@ bool LoopVectorizationLegality::canVectorizeInstrs() {
       if (CI && !VFDatabase::getMappings(*CI).empty())
         VecCallVariantsFound = true;
 
+      auto canWidenInstruction = [this](Instruction const &Inst) {
+        Type *InstTy = Inst.getType();
+        if (isa<CallInst>(Inst) && isa<StructType>(InstTy) &&
+            canWidenCallReturnType(InstTy)) {
+          StructVecCallFound = true;
+          // For now, we can only widen struct values returned from calls where
+          // all users are extractvalue instructions.
```

Comment:
This doesn't seem right yet?

Reply:
I think its right in terms of what `LoopVectorizationLegality` can see as vectorizable

Reply:
Hm I see, would probably be good to clarify the comment, as the current wording seems to imply we can already widen it (which to me reads as we can create wide vector recipes for them)

Reply:
Rephased the comment (and added a TODO around the flag, which would help clarify too).